src/Form/tests/FormAutosuggest.test.jsx

```diff
@@ -25,141 +14,179 @@ function FormAutosuggestWrapper(props) {
   );
 }
 
-describe('FormAutosuggest', () => {
-  afterEach(() => {
-    jest.clearAllMocks();
-  });
-
-  const onSelected = jest.fn();
-  const onClick = jest.fn();
-
-  const container = mount(
+function FormAutosuggestTestComponent(props) {
+  return (
     <FormAutosuggestWrapper
       name="FormAutosuggest"
       floatingLabel="floatingLabel text"
       helpMessage="Example help message"
       errorMessageText="Example error message"
-      onSelected={onSelected}
+      onSelected={props.onSelected}
     >
       <FormAutosuggestOption>Option 1</FormAutosuggestOption>
-      <FormAutosuggestOption onClick={onClick}>Option 2</FormAutosuggestOption>
+      <FormAutosuggestOption onClick={props.onClick}>Option 2</FormAutosuggestOption>
       <FormAutosuggestOption>Learn from more than 160 member universities</FormAutosuggestOption>
-    </FormAutosuggestWrapper>,
+    </FormAutosuggestWrapper>
   );
+}
 
-  describe('render behavior', () => {
-    it('renders component without error', () => {
-      mount(<FormAutosuggestWrapper />);
-    });
+FormAutosuggestTestComponent.defaultProps = {
+  onSelected: jest.fn(),
+  onClick: jest.fn(),
+};
 
-    it('render without loading state', () => {
-      expect(container.exists('.pgn__form-autosuggest__dropdown-loading')).toBe(false);
-      expect(container.props().isLoading).toBeUndefined();
-    });
+FormAutosuggestTestComponent.propTypes = {
+  /** Specifies onSelected event handler. */
+  onSelected: PropTypes.func,
+  /** Specifies onClick event handler. */
+  onClick: PropTypes.func,
+};
 
-    it('render with loading state', () => {
-      const wrapper = mount(<FormAutosuggestWrapper isLoading />);
+describe('render behavior', () => {
+  it('renders component without error', () => {
+    render(<FormAutosuggestWrapper />);
+  });
 
-      expect(wrapper.exists('.pgn__form-autosuggest__dropdown-loading')).toBe(true);
-      expect(wrapper.props().isLoading).toBe(true);
-    });
+  it('renders without loading state', () => {
+    const { container } = render(<FormAutosuggestTestComponent />);
+    expect(container.querySelector('.pgn__form-autosuggest__dropdown-loading')).toBeNull();
```

Comment:
Is there a way to avoid using the class name implementation detail in this test?

For example, if the component renders a `Spinner` with the supplied `screenReaderText` prop, could this test assert that the `Spinner` is not rendered based on a `data-testid` attribute versus relying on the class name that could change at any time. If the class name did change down the road, this test would still pass.

Reply:
Fixed in:
https://github.com/openedx/paragon/compare/299441986865ab87377629aa4d99f13ed8fc47d7..6c9a96ea3e2ba2ef4ab37d5f266b4d7205f5c718